src/protocol/attribution/aggregate_credit.rs

```diff
@@ -253,19 +250,21 @@ async fn sort_by_breakdown_key<F: Field>(
     num_multi_bits: u32,
 ) -> Result<Vec<CappedCreditsWithAggregationBit<F>>, Error> {
     // TODO: Change breakdown_keys to use XorReplicated to avoid bit-decomposition calls
-    let breakdown_keys = transpose(
-        &bit_decompose_breakdown_key(ctx.narrow(&Step::BitDecomposeBreakdownKey), input).await?,
-    );
-
     // We only need to run a radix sort on the bits used by all possible
     // breakdown key values.
     let valid_bits_count = u128::BITS - (max_breakdown_key - 1).leading_zeros();
 
-    let sort_permutation = generate_permutation_and_reveal_shuffled(
-        ctx.narrow(&Step::GeneratePermutationByBreakdownKey),
-        &breakdown_keys[..valid_bits_count as usize],
+    let bit_decompose_breakdown_key =
+        bit_decompose_breakdown_key(ctx.narrow(&Step::BitDecomposeBreakdownKey), input).await?;
+    let breakdown_keys = split_into_multi_bit_slices(
+        &bit_decompose_breakdown_key,
```

Comment:
I think @taikiy is/has removed this.

Reply:
#418. Not merged yet but it will be removed.